Add browserContext.addInitScript documentation (#1527)
grafana/k6-docs: docs/sources/next/javascript-api/k6-experimental/browser/browsercontext/addinitscript.md

---
title: 'addInitScript()'
description: 'Adds an init script.'
---

# addInitScript()

Adds a script which will be evaluated in one of the following scenarios:

> - Whenever a page is created in the browser context or is navigated.

Author: Sure, done in 33bbb146fc89d497d39ecaaf0ec2854024d0d1ff.

> - Whenever a child frame is attached or navigated in any page in the browser context. In this case, the script is evaluated in the context of the newly attached frame.

Author: Sure, done in 33bbb146fc89d497d39ecaaf0ec2854024d0d1ff.

> The script is evaluated after the document is created but before any of its scripts were run. This is useful to amend the JavaScript environment, e.g. to override `Math.random`.

Reviewer: ```suggestion
The script is evaluated after the document is created but before any of its scripts are run. This is useful to amend the JavaScript environment, for example, to override `Math.random`.
```
